### PR TITLE
Population genetics model section

### PR DESCRIPTION
**Pull request by apragsdale**
This is a rough start to that section. This might be too much detail (or not enough), or maybe these details should go somewhere else. Let me know what you all think.

### CODE REVIEW

File: paper.tex [hunk 1]
```diff
@@ -207,34 +207,86 @@ \section*{Demes}
...
+specification; and the parser, which is responsible for transforming the former
+into the latter.
+
+\subsection*{Population genetics model}
+
+In Demes, demographic models consist of one or more interacting populations (or
+``demes"; to avoid confusion with the name of the specification itself we will
+use the term ``population'' in this discussion, in the understanding that the
+terms are interchangeable for our purposes). Populations are defined as a
+collection of exchangable individuals that exist a specified period of time,
```
Comment (grahamgower) [inline paper.tex]:
```suggestion
collection of exchangable individuals that exist for a specified period of time,
```

File: paper.tex [hunk 2]
```diff
@@ -207,34 +207,86 @@ \section*{Demes}
...
+following a well-defined set of rules regarding population sizes, mating
+systems, etc. Ancestor/descendent relationships between populations can be
```
Comment (grahamgower) [inline paper.tex]:
```suggestion
systems, etc. Ancestor/descendant relationships between populations can be
```

File: paper.tex [hunk 3]
```diff
@@ -207,34 +207,86 @@ \section*{Demes}
...
+defined, as well as continuous or instantaneous migration between coexisting
+populations.
+
+Population and event times are written as units in the past, so that time zero
+corresponds to the final generation or ``now'', and event times in the past are
+values greater than zero with larger values for events that occur in the more
+distant past. While a natural specification for time units is in generations,
+other time units are allowed, such as years. In the case the the time units are
```
Comment (grahamgower) [inline paper.tex]:
```suggestion
other time units are allowed, such as years. In the case that the time units are
```

File: paper.tex [hunk 4]
```diff
@@ -207,34 +207,86 @@ \section*{Demes}
...
+offspring in the newly generated population. If more than one ancestor is
+specified, the proportions of ancestry from each contributing population must
+be provided, and those proportions must sum to one. In this case, parents are
+chosen randomly from each ancestral population with probability given by those
+proportions. If no ancestors are specified, the population is assumed to have
+start time equal to infinity, as individuals are not assumed to spontaneously
+spring into existence.
+
+Within populations, population sizes and mating systems are specified within
+epochs. Epochs are non-overlapping intervals that cover the entire existence
```
Comment (grahamgower) [inline paper.tex]:
```suggestion
epochs. Epochs are non-overlapping intervals of time that cover the entire existence
```

File: paper.tex [hunk 5]
```diff
@@ -207,34 +207,86 @@ \section*{Demes}
...
+interval of the population. Each epoch specifies the population size over that
+interval, which can be a constant value or function defined by start and end
+sizes that must remain positive.  If an epoch has a start time of infinity, the
+population size for that epoch must be constant. Epochs can also specify
+parameters for nonrandom mating, such as selfing or cloning rates, which give
+the probability that offspring are generated from one generation to the next by
+self-fertilization or cloning of an indivual \textbf{need to precisely specify
```
Comment (grahamgower) [inline paper.tex]:
```suggestion
self-fertilization or cloning of an individual \textbf{need to precisely specify
```

File: paper.tex [hunk 6]
```diff
@@ -207,34 +207,86 @@ \section*{Demes}
...
+once this is determined}.
+
+Finally, individuals in a population may have parents from a different
+population through migrations. These can be defined as continuous migration
+rates over intervals that populations overlap in time or through instantaneous
```
Comment (grahamgower) [inline paper.tex]:
```suggestion
rates over time intervals in which the populations coexist or through instantaneous
```

File: paper.tex [hunk 7]
```diff
@@ -207,34 +207,86 @@ \section*{Demes}
...
+(or pulse) migration events at a given time. Continuous migration rates are
+defined as the probability that parents in the ``destination`` population are
+chosen from the ``source`` population.  Migration rates are thus per-generation
```
Comment (grahamgower) [inline paper.tex]:
```suggestion
defined as the probability that parents in the ``destination'' population are
chosen from the ``source'' population.  Migration rates are thus per generation
```

File: paper.tex [hunk 8]
```diff
@@ -207,34 +207,86 @@ \section*{Demes}
...
+and must be less than one. Furthermore, if more than one source population have
+continuous migration into the same destination population, the sum of those
+migration rates must also be less than one, as rates define probabilities. The
+probability that parents come from the same population is just one minus the
+some of incoming migration rates. On the other hand, pulse migration events
```
Comment (grahamgower) [inline paper.tex]:
```suggestion
sum of incoming migration rates. On the other hand, pulse migration events
```

File: paper.tex [hunk 9]
```diff
@@ -207,34 +207,86 @@ \section*{Demes}
...
+specify the instantaeous replacement of a given fraction of individuals in a
+destination population by individuals with parents from a source population.
+The fraction must be between zero and one, and if more than one source
+population has a pulse migration event to the same destination population and
+the same time, those replacement events are applied sequentially in the order
+that they are specified in the model.
```
Comment (grahamgower) [inline paper.tex]: This is not quite sufficient, if one considers two pulses like A->B, B->C. The order here matters, but the pulse destination is different in each case.
```suggestion
The fraction must be between zero and one, and if more than one pulse is defined
for the same time, those replacement events are applied sequentially in the order
that they are specified in the model.
```

**Review by grahamgower — APPROVED** [review]
This is great! We should definitely copy this into the spec. I'm not sure about having it in the paper, but probably it's a good idea so readers know what the model is without looking further.

**Review by jeromekelleher — APPROVED** [review]
This is great, thanks @apragsdale!

I agree with Graham, in that this is more in line with what we want for the specification rather than the paper. I think what we want here is a shortened abstract, which gives the essential ideas, without getting into gnarly details.

In the spec then, we go full on for all the gnarly details. The idea is that the spec is definitive, so that's where people go for details. The paper is about giving a high-level understanding of the model itself and explaining the rationale behind decisions.

One suggestions as to how to go forward would be to merge this, with a note that it needs to be abstracted out a bit, and then we copy the text as the starting point for the same section in the spec?

**Comment by grahamgower** [comment]
> One suggestions as to how to go forward would be to merge this, with a note that it needs to be abstracted out a bit, and then we copy the text as the starting point for the same section in the spec?

Yeah, I think that's a good way forward here.

**Comment by apragsdale** [comment]
Ok great - I agree that it's too much detail for the paper, but just wanted to jot down what I thought was the main information for the pop gen model specification. Thanks for the careful read-through @grahamgower. I'll clean it up later today so hold of on merging for the moment.

**Comment by jeromekelleher** [comment]
How's this looking @apragsdale? It would be great to get this text into the specification, so we've something to get working with.

**Comment by apragsdale** [comment]
Just getting to this now.. was out of the house/office the last two days

**Comment by apragsdale** [comment]
Thanks for the detailed comments catching all my embarrassing typos, @grahamgower. I'm happy to merge this, and we can rearrange and further pare it down later on.

**Comment by jeromekelleher** [comment]
Hi @apragsdale - would you mind copying this text over to the same section in the spec please? I don't want to steal your git-cred by doing it myself!

**Pull request merged**